src/redux/wallets.js

```diff
@@ -112,13 +127,48 @@ export const walletsSetSelected = wallet => async dispatch => {
   });
 };
 
-export const isCreatingAccount = val => dispatch => {
+export const setIsWalletLoading = val => dispatch => {
   dispatch({
     payload: val,
-    type: WALLETS_SET_IS_CREATING_ACCOUNT,
+    type: WALLETS_SET_IS_LOADING,
   });
 };
 
+export const setWalletBackedUp = (
+  walletId,
+  method,
+  backupFile = null
+) => async (dispatch, getState) => {
+  const { wallets, selected } = getState().wallets;
+  const newWallets = { ...wallets };
+  newWallets[walletId].backedUp = true;
+  newWallets[walletId].backupType = method;
+  if (backupFile) {
+    newWallets[walletId].backupFile = backupFile;
+  }
+  newWallets[walletId].backupDate = Date.now();
+  await dispatch(walletsUpdate(newWallets));
```

Comment:
just FYI: we do awaits on walletsUpdate, but it itself doesn't await the saveAllWallets (not sure if we care)